modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java

```diff
@@ -0,0 +1,228 @@
+    private final String activeTenant;
+    private final AutomationContext context;
+    private final TestUserMode testUserMode;
+
+    private static final String SYSTEM_SCOPE = "SYSTEM";
+
+    @DataProvider(name = "configProvider")
+    public static Object[][] configProvider() {
+
+        return new Object[][]{{TestUserMode.SUPER_TENANT_ADMIN}, {TestUserMode.TENANT_ADMIN},
+                {TestUserMode.TENANT_USER}};
+    }
+
+    @Factory(dataProvider = "configProvider")
+    public SystemScopePermissionValidationTestCase(TestUserMode userMode) throws Exception {
+
+        context = new AutomationContext("IDENTITY", userMode);
+        this.username = context.getContextTenant().getTenantAdmin().getUserName();
+        this.userPassword = context.getContextTenant().getTenantAdmin().getPassword();
+        this.activeTenant = context.getContextTenant().getDomain();
+        this.testUserMode = userMode;
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        backendURL = context.getContextUrls().getBackEndUrl();
+        logManger = new AuthenticatorClient(backendURL);
+        sessionCookie = logManger.login(username, userPassword, context.getInstance().getHosts().get("default"));
+        tenantInfo = context.getContextTenant();
+        userInfo = tenantInfo.getContextUser();
+        appMgtclient = new ApplicationManagementServiceClient(sessionCookie, backendURL, null);
+        adminClient = new OauthAdminClient(backendURL, sessionCookie);
+
+        setSystemproperties();
+        client = HttpClientBuilder.create().build();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+
+        appMgtclient.deleteApplication(SERVICE_PROVIDER_NAME);
+        adminClient.removeOAuthApplicationData(consumerKey);
+        client.close();
+        logManger = null;
+        consumerKey = null;
+        accessToken = null;
+    }
+
+    @Test(groups = "wso2.is", description = "Check Oauth2 application registration")
+    public void testRegisterApplication() throws Exception {
+
+        OAuthConsumerAppDTO appDto = createApplication();
+        Assert.assertNotNull(appDto, "Application creation failed.");
+
+        consumerKey = appDto.getOauthConsumerKey();
+        Assert.assertNotNull(consumerKey, "Application creation failed.");
+
+        consumerSecret = appDto.getOauthConsumerSecret();
+        Assert.assertNotNull(consumerSecret, "Application creation failed.");
+    }
+
+    @Test(groups = "wso2.is", description = "Send authorize user request and get access token", dependsOnMethods = "testRegisterApplication")
+    public void testGetAccessToken() throws Exception {
+
+        List<NameValuePair> urlParameters = new ArrayList<>();urlParameters.add(new BasicNameValuePair("grantType",
+                OAuth2Constant.OAUTH2_GRANT_TYPE_CLIENT_CREDENTIALS));
+        urlParameters.add(new BasicNameValuePair("consumerKey", consumerKey));
```

Comment:
Don't we have enums/constants for these "consumerKey" etc?

Reply:
We have a constant for `consumerKey`, but not for other attributes like `consumerSecret`, `accessEndpoint` etc. Also, most of the tests have straightaway used these keys(without using any constant or enums) when they send similar types of requests.